### PR TITLE
added baseline implementation for all remaining api endpoints

### PR DESCRIPTION
**Pull request by friedliver**

### CODE REVIEW

**Review by Guillar1 — APPROVED** [review]

File: src/reporting/reportingClient.ts [hunk 1]
```diff
@@ -37,24 +58,132 @@ const prefixUrl = (baseUrl?: string, prefix?: string) => {
...
+   * @summary OData Entity
+   * @param {string} reportId The Report Id.
+   * @param {ODataItem} odataItem Reference to a table exported to your Report. Use {@link getODataReport()} to fetch a list of ODataItems in the report.
+   * @param {string} accessToken OAuth access token with scope &#x60;insights:read&#x60;
+   * @memberof ReportingClient
+   */
+  public async getODataReportEntity(accessToken: AccessToken, reportId: string, odataItem: ODataItem) {
+    const segments = odataItem?.url?.split('/');
+    if (segments?.length !== 3) return undefined;
```
Comment (bingjiez) [inline src/reporting/reportingClient.ts:96]: separate lines

File: src/reporting/reportingClient.ts [hunk 2]
```diff
@@ -100,16 +292,46 @@ export class ReportingClient {
...
+    const reportMappings: Array<ReportMapping> = [];
+
+    let response: ReportMappingCollection;
+    let continuationToken: string | undefined;
+
+    do {
+      response = await this._reportsApi.getReportMappings(
+        reportId,
+        accessToken,
+        undefined,
```
Comment (bingjiez) [inline src/reporting/reportingClient.ts:312]: Do we want to support top?

File: src/reporting/reportingClient.ts [hunk 3]
```diff
@@ -268,20 +617,57 @@ export class ReportingClient {
...
-      groupId,
-      accessToken
-    );
+    const properties: Array<GroupProperty> = [];
+
+    let response: GroupPropertyCollection;
+    let continuationToken: string | undefined;
+
+    do {
+      response = await this._mappingsApi.getGroupproperties(
```
Comment (bingjiez) [inline src/reporting/reportingClient.ts:641]: Not following camel case

File: src/reporting/reportingClient.ts [hunk 4]
```diff
@@ -298,6 +684,16 @@ export class ReportingClient {
...
...with valid Bearer token for scope `insights:modify`.    For more documentation on authorization and how to get access token visit [OAUTH2 Authorization](https://developer.bentley.com/apis/overview/authorization/) page.    ### Authorization    User must have `imodels_write` permission(s) assigned at the Project level. iModel specific permissions may also be applied at the iModel level if iModel level permissions are enabled.    Alternatively the user should be an Organization Administrator for the Organization that owns a given Project or iModel.    An Organization Administrator must have at least one of the following roles assigned in User Management: Account Administrator, Co-Administrator, or CONNECT Services Administrator. For more information about User Management please visit our Bentley Communities [Licensing, Cloud, and Web Services](https://communities.bentley.com/communities/other_communities/licensing_cloud_and_web_services/w/wiki/50711/user-management-2-0) wiki page.    ### Rate limits    All iTwin Platform API operations have a rate limit. For more documentation on that visit [Rate limits and quotas](https://developer.bentley.com/apis/overview/rate-limits/) page.    ---
```
Comment (bingjiez) [inline src/reporting/reportingClient.ts:688]: Shall we put repeated docs in a single region like top of all mappings and point to the place at the end

**Review by bingjiez — APPROVED** [review]

File: src/reporting/reportingClient.ts (hunk 2 above)
Comment (arnobmallickbsw) [reply src/reporting/reportingClient.ts:312]: yes, but not right now
leaving at minimum implementation and we will fix the client properly soon

File: src/reporting/reportingClient.ts (hunk 3 above)
Comment (arnobmallickbsw) [reply src/reporting/reportingClient.ts:641]: this is from the generated api
we will fix this later when we get rid of the generated client

File: src/reporting/reportingClient.ts (hunk 4 above)
Comment (arnobmallickbsw) [reply src/reporting/reportingClient.ts:688]: I added the JSDocs so that developers will have intellisense on the ReportingClient

**Review by arnobmallickbsw — APPROVED** [review]

File: src/reporting/reportingClient.ts (hunk 1 above)
Comment (arnobmallickbsw) [reply src/reporting/reportingClient.ts]: fixed

**Pull request merged**